event_model/__init__.py

```diff
@@ -1620,6 +1702,54 @@ def compose_resource(*, spec, root, resource_path, resource_kwargs,
         partial(compose_datum_page, resource=doc, counter=counter))
 
 
+def compose_stream_datum(*, stream_resource, stream_name, counter, datum_kwargs,
+                         event_count=1, event_offset=0, validate=True):
+    resource_uid = stream_resource['uid']
```

Comment:
Should we add a check here that the resource document actually know about the stream we are trying to make?

On one hand YOLO why does the resource need to know what streams it will contain?  On the other hand, the resource document currently advertises the expected streams, those we should make sure we do not generate a stream of documents that is not internally consistent.

History has shown that if we have a "MUST" (or "SHOULD") requirement that is not enforced by code, we will end up with reams of documents that are incorrect.  The primary example here is all of the issues we have been cleaning up with the shapes!

Reply:
Adjustments made in source and testing for validation in https://github.com/bluesky/event-model/pull/222/commits/9082b82c92e0c7833555dadb12c3efaf17b478fe